subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ComponentMetadataRulesIntegrationTest.groovy

```diff
@@ -184,7 +214,7 @@ dependencies {
                 }
             }
 
-            resolve.doLast { assert rulesInvoked == [ '1.0', '1.0', '1.0', '1.0', '1.0' ] }
+            resolve.doLast { assert rulesInvoked == [ '1.0', '1.0', '1.0', '1.0', '1.0', '1.0' ] }
```

Comment:
:x: There's something fishy here. If this test passes, it means we're not properly isolating parameters. The rule shouldn't access the same list, or it means we cannot properly isolate them. Said differently, it has a side effect. The goal of the `params` block is to initialize a rule, but it shouldn't be able to mutate the state of the project, which is implicitly what it does here.